modules/microshift-nw-create-http-based-route.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * microshift_networking/microshift-configuring-routes.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-nw-creating-a-route_{context}"]
+= Creating an HTTP-based route
+
+A route allows you to host your application at a public URL. It can either be secure or unsecured, depending on the network security configuration of your application. An HTTP-based route is an unsecured route that uses the basic HTTP routing protocol and exposes a service on an unsecured application port.
+
+The following procedure describes how to create a simple HTTP-based route to a web application, using the `hello-microshift` application as an example.
+
+.Prerequisites
+* You installed the OpenShift CLI (`oc`).
+* You have root access to your {microshift-short} cluster.
+* You have a web application that exposes a port and a TCP endpoint listening for traffic on the port.
+
+.Procedure
+
+. Create a service called `hello-microshift` by running the following command:
++
+[source,terminal]
+----
+$ oc expose pod hello-microshift -n $namespace
+----
+
+. Create an unsecured route to the `hello-microshift` application by running the following command:
++
+[source,terminal]
+----
+$ oc expose svc/hello-microshift --hostname=example.com
+----
+
+.Verification
+
+* To verify that the `route` resource that you created, run the following command:
++
+[source,terminal]
+----
+$ oc get routes -o yaml <name of resource> <1>
+----
+<1> In this example, the route is named `hello-microshift`.
+
+.Sample YAML definition of the created unsecured route:
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: hello-microshift
+spec:
+  host: example.com
+  port:
+    targetPort: 8080 <1>
+  to:
+    kind: Service
+    name: hello-microshift
+----
+<1> `targetPort` is the target port on pods that is selected by the service that this route points to.
+
++
+[NOTE]
+====
+To display your default ingress domain, run the following command:
+[source,terminal]
+----
+$ oc get ingresses.config/cluster -o jsonpath={.spec.domain}
```

Comment:
running this line, I get a messages saying the server doesn't have a resource type "ingress"

`[redhat@microshift-dev ~]$ oc get ingresses.config/cluster -o jsonpath={.spec.domain}
error: the server doesn't have a resource type "ingresses"
`

Reply:
@pacevedom Hmm, do we need to make a CR?

Reply:
This is not a resource MicroShift supports. We do support k8s ingresses but this example does not change any of them.
```$ oc api-resources | grep ingress
ingressclasses                                          networking.k8s.io/v1                false        IngressClass
ingresses                           ing                 networking.k8s.io/v1                true         Ingress```

Reply:
So how would we display the default ingress domain in microshift? Something like, `$ oc get pod -n openshift-ingress router-default-5b75594b4-228z7 -o jsonpath='{.status.podIPs}'`?

Reply:
Ah now I understand. We dont have the api for the [ingress](https://docs.openshift.com/container-platform/4.16/rest_api/config_apis/ingress-config-openshift-io-v1.html), so we dont really have a default domain.
The certificate we use in the router for https includes this for the hostname: https://github.com/openshift/microshift/blob/main/pkg/cmd/init.go#L233
And each route may define their own hostname too.